pkg/block/fetcher.go

```diff
@@ -37,7 +37,36 @@ type syncMetrics struct {
 	syncFailures prometheus.Counter
 	syncDuration prometheus.Histogram
 
-	synced *extprom.TxGaugeVec
+	synced   *extprom.TxGaugeVec
+	modified *extprom.TxGaugeVec
+}
+
+func (s *syncMetrics) submit() {
+	if s == nil {
+		return
+	}
+
+	if s.synced != nil {
```

Comment:
Why those nils are needed? Can we just do 
`s.synced.Submit`
`s.modified.Submit` always?

Reply:
Of course, I was just being precautions.